test/extended/util/test.go

```diff
@@ -139,6 +142,14 @@ func verifyTestSuitePreconditions() {
 		// uid don't get forced into a range (mimics upstream
 		// behavior)
 		addE2EServiceAccountsToSCC(c, namespaces, "anyuid")
+
+		// The intra-pod test requires that the service account have
+		// permission to retrieve service endpoints.
+		osClient, _, err := configapi.GetOpenShiftClient(KubeConfigPath())
+		if err != nil {
+			FatalErr(err)
+		}
+		addRoleToE2EServiceAccounts(osClient, namespaces, bootstrappolicy.EditRoleName)
```

Comment:
Bound their permissions is my preference.  I thought endpoints were allowed to viewers since EAP needs it?  Is EAP setting up a custom role (I hope not, I hope not, I hope not...)